app/src/main/java/org/connectbot/ConsoleActivity.java

```diff
@@ -169,20 +166,13 @@ public void onServiceConnected(ComponentName className, IBinder service) {
 			}
 
 			// create views for all bridges on this service
-			for (TerminalBridge bridge : bound.getBridges()) {
-
-				final int currentIndex = addNewTerminalView(bridge);
-
-				// check to see if this bridge was requested
-				if (bridge == requestedBridge)
-					requestedIndex = currentIndex;
-			}
+			adapter.notifyDataSetChanged();
+			requestedIndex = bound.getBridges().indexOf(requestedBridge);
 
-			setDisplayedTerminal(requestedIndex);
+			pager.setCurrentItem(requestedIndex == -1 ? 0 : requestedIndex);
```

Comment:
I think this is missing an `updatePromptVisible`, etc. Perhaps you should wrap all that in a new `setDisplayedTerminal` that calls `pager.setCurrentItem(…)` in addition to `updatePromptVisible()` and `updateEmptyVisible()`.